src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -64,37 +64,49 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                 return Task.FromResult<IDictionary<string, object>>(
                     new Dictionary<string, object>(StringComparer.Ordinal));
             }
-            else
+            else if (actionDescriptor.BoundProperties.Count == 0)
             {
                 return BindActionArgumentsCoreAsync(
                     context,
                     controller,
                     actionDescriptor);
             }
+            else
+            {
+                return BindActionArgumentsAndPropertiesCoreAsync(
+                    context,
+                    controller,
+                    actionDescriptor);
+            }
         }
 
-        private async Task<IDictionary<string, object>> BindActionArgumentsCoreAsync(
+        private async Task<IDictionary<string, object>> BindActionArgumentsAndPropertiesCoreAsync(
```

Comment:
Consider dropping the `Core` term. The other one helped to distinguish it from `BindActionArgumentsAsync` - this one just makes it verbose.